### ISSUE TITLE
Import must always use newest package and add require() function

### ISSUE THREAD

**Issue by NiklasRosenstein** [NiklasRosenstein/pypack#2]
The `PyPackImporter` should always prefer to import a module from the newest pack it can find (determined by the version number). If two Python packs with the same version number, that can both provide the requested module, are found, the order in which the packs where added to the `PyPackImporter.cache` decides which one to use. Packs that have no version number specified should be considered the least optimal packs.

A function `pypack.require(package_spec)` should be added that will load a PyPack that matches the specification *package_spec*. The package specification should be a string that defines the package or module name that is to be loaded and optionally the version number that should be loaded, basically the same as [PIP requirements][]. If the specification would match multiple packs, the pack with the newest version must be used. The `require()` method should be able to load a PyPack completely without replacing an already existing module. It might be necessary to temporarily shift out conflicting modules from the global module cache in order to properly load a PyPack with `require()`. If the package specification would match the newest version that would also be imported with a normal `import` statement, that exact module may be re-used from the global module cache. The `require()` method must return the root module of the imported package.

To sum it up, the following code should execute without problems, given that the PyPack `my.package` in versions `0.8a`, `0.7`, `0.5c`, `0.5a` and `0.4f can be found:

    import pypack
    import my.package                      # loads 0.8a
    my07 = require('my.package==0.7')      # loads 0.7
    my05 = require('my.package<=0.6>0.4')  # loads 0.5c
    

  [PIP requirements]: https://pip.pypa.io/en/latest/user_guide.html#requirements-files

**Comment by NiklasRosenstein** [comment]
Need #2 before this can be implemented.
